portal/get_patient_info.php

```diff
@@ -174,22 +174,22 @@
     if (empty($userData)) {
         $logit->portalLog('login attempt', '', ($_POST['uname'] . ':not active patient'), '', '0');
         OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
-        header('Location: ' . $landingpage . '&w');
+        header('Location: ' . $landingpage . '&w=inactive');
```

Comment:
from a security perspective, it's generally best to not return any info on a bad login

Reply:
as an example, in core OpenEMR login, not only do we not return any specific messages, but we even make the time to check the hash the same (ie. run a fake hash if uses does not exist) so bad actors can't figure things out with timing attacks.

Reply:
tbh there has been multiple times where this login mechanism didn't work at all on me and I entered everything correctly (and it's only the portal doing this). I felt like adding this for debugging reasons yet since it can destroy security I'll revert it and create a separate issue on it.

Reply:
for debugging, rec ensuring the issues are always sent to the portal log (had the same issue in core openemr login debugging, so basically sent specific issues to the openemr log to be able to debug issues; since like portal there are several layers where authentication breask)

Reply:
> tbh there has been multiple times where this login mechanism didn't work at all on me

this worries me! I use portal a lot. creating and using new credentials, old credential etc. we need to keep an eye on if you're having issues.

What are we going to do about RTL Header? I say go ahead and modify Header class for portal include conditional.

Reply:
@sjpadgett What worries me a lot is that when the login does fail like this it doesn't tell you why it failed or give u a php or sql error. It also happens very sporadically and can even happen on a fresh install.

Reply:
@sjpadgett As far as the RTL header goes adding a method that turns off the regular stylesheet when RTL stylesheet is active is the way to go here

Reply:
> What worries me a lot is that when the login does fail like this it doesn't tell you why it failed or give u a php or sql error

Prob a JS error then but agree with Brady, never a good idea to give any failure hints to user. However, I thought we had logging in place for auth errors where they fail. Perhaps reviewing your logs will tell something.


> turns off the regular stylesheet when RTL stylesheet is active is the way to go here

For portal only or all styles?